### DISCUSSION TITLE
Create autoEvent fails

### DISCUSSION THREAD

**Discussion by IkerDC** [edgexfoundry/community#163] [Q&A]
**TL;DR**:

For our use case, we have several devices with running autoEvents. From an app service in the North, we are subscribed to events of those devices. When an autoEvent is received, we do something. However, **we also need to know when the autoEvent has failed**, because in that case we have to do something else.

**Question:**

Is there a way to know when a device autoEvent has failed?

**Comment by mglmc** [comment]
As Iker said, there is a problem here with using autoevents for pulling actions for protocols that don't have last will or something similar, such as Modbus or REST. In these cases, it is useful to know when the device is down, so that we can take appropriate action.

We have been working on a solution to this problem, and we have found a way to notify the north side when an autoevent has failed. We do this by sending an ASYNC message.

For example, for Modbus devices, when an error happens in the connection, we send an async event with the resource `failedConnection`

![code_asyncMsg](https://github.com/edgexfoundry/community/assets/20265876/365c6cd3-65e4-4766-b6d1-d722e804274b)

![code_asyncMsg_HandleReadCommands](https://github.com/edgexfoundry/community/assets/20265876/6f8c9423-7e3f-4c32-9d6f-59cbfe943f98)

We believe that this is a useful feature that could be helpful for a lot of EdgeX Foundry developers. We are wondering if it would be possible to add this feature to the official device service, or if it could be implemented as an example device service in another place.

Please let us know your thoughts.

**Comment by cloudxxx8** [comment]
@lenny-intel we got similar request from other users.  When the southbound or northbound connection go wrong from Device Service and App Service, we may send out a system event as an alert.  Ideally, Notifications Service should be able to receive and process this alret.  Maybe we can plan the system event in the next release.
@IkerDC for now, you may adopt @mglmc 's suggestion to implement something in your device service.

**Reply by lenny-goodell** [reply]
@cloudxxx8 , Sounds like a good topic for discussion for next Core WG meeting.

**Reply by cloudxxx8** [reply]
@lenny-intel oh, I missed this today.
@jumpingliu please put this topic into the agenda of our next Core WG meeting

**Reply by jumpingliu** [reply]
@cloudxxx8 ok added.

**Reply by mglmc** [reply]
Glad to hear that. I really think that is a mandatory FTR for a lot of use cases. Meanwhile, we'll be using this workaround.
